MobiFlight/Board.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MobiFlight
+{
+    public class AvrDudeSettings
+    {
+        /// <summary>
+        /// AvrDude device type for the device.
+        /// </summary>
+        public String Device;
+        /// <summary>
+        /// Baud rate to use with AvrDude
+        /// </summary>
+        public String BaudRate;
+        /// <summary>
+        /// AvrDude programmer to use for the device.
+        /// </summary>
+        public String Programmer;
+        /// <summary>
+        /// Base name for firmware files. The final filename is of the form {FirmwareBaseName}_{Version}.hex.
+        /// </summary>
+        public String FirmwareBaseName;
+        public string GetFirmwareName(string latestFirmwareVersion)
+        {
+            return $"{FirmwareBaseName}_{latestFirmwareVersion.Replace('.', '_')}.hex";
+        }
+    }
+
+    public class Board
+    {
+        /// <summary>
+        /// Maximum number of outputs supported by the board.
+        /// </summary>
+        public int MaxOutputs = 0;
+        /// <summary>
+        /// Maximum number of buttons supported by the board.
+        /// </summary>
+        public int MaxButtons = 0;
+        /// <summary>
+        /// Maximum number of LED segments supported by the board.
+        /// </summary>
+        public int MaxLedSegments = 0;
+        /// <summary>
+        /// Maximum number of encoders supported by the board.
+        /// </summary>
+        public int MaxEncoders = 0;
+        /// <summary>
+        /// Maximum number of steppers supported by the board.
+        /// </summary>
+        public int MaxSteppers = 0;
+        /// <summary>
+        /// Maximum number of servos supported by the board.
+        /// </summary>
+        public int MaxServos = 0;
+        /// <summary>
+        /// Maximum number of I2C LCDs supported by the board.
+        /// </summary>
+        public int MaxLcdI2C = 0;
+        /// <summary>
+        /// Maximum number of analog inputs supported by the board.
+        /// </summary>
+        public int MaxAnalogInputs = 0;
+
+        /// <summary>
+        /// True if multiple attempts should be made when connecting to the board.
+        /// </summary>
+        public Boolean ExtraConnectionRetry;
+        /// <summary>
+        /// True if the board supports loading firmware via MobiFlight.
+        /// </summary>
+        public Boolean CanInstallFirmware;
```

Comment:
This property is new and gets used elsewhere to skip firmware checks if `false`. Makes it really easy to have a board definition file for non-Arduino boards where the board should be detected and used by MobiFlight but flashing isn't supported.

Reply:
Cool idea.